src/master_node/services/websocket_server_service.py

```
@@ -0,0 +1,71 @@
+            print(f"Worker {worker_id} disconnected with an error: {e}")
+        finally:
+            if str(worker_id) in self.connections:
+                del self.connections[str(worker_id)]
+                print(f"Cleaned up connection for worker {worker_id}")
+    
+    async def handle_worker_message(self, worker_id: UUID, message: dict[str, Any]):
+        message_type = message.get("type")
+        if not message_type:
+            print(f"Missing 'type' in message from worker {worker_id}: {message}")
+            ws = self.connections.get(str(worker_id))
+            if ws:
+                await ws.send_json({"error": "Missing 'type' in message"})
+            return
+        
+        if message_type == "heartbeat":
+            print(f"Heartbeat from worker {worker_id}")
+        elif message_type == "task_result":
+            print(f"Task result from worker {worker_id}: {message.get('data')}")
+        elif message_type == "status_update":
+            print(f"Status update from worker {worker_id}: {message.get('status')}")
+        else:
+            print(f"Unknown message type from worker {worker_id}: {message}")
+            ws = self.connections.get(str(worker_id))
+            if ws:
+                await ws.send_json({"error": "Unknown message type"})
+    
+    async def send_to_worker(self, worker_id: UUID, data: dict[str, Any]):
+        worker_key = str(worker_id)
+        ws = self.connections.get(worker_key)
+        if ws:
+            try:
+                await ws.send_json(data)
+                print(f"Sent to {worker_id}: {data}")
+                return True
+            except Exception as e:
+                print(f"Failed to send to worker {worker_id}: {e}")
+                
+                if worker_key in self.connections:
+                    del self.connections[worker_key]
+                return False
+        else:
+            print(f"Worker {worker_id} not connected")
+            return False
```

For the check if worker node is connected first convert this nested if to a guard clause then instead of returning true or false just raise an exception if an error has occurred.